documentation/proposals/Proposal - Silk.NET 3.X Generic Maths.md

```diff
@@ -0,0 +1,884 @@
+    /// <param name="y">The value to assign to the <see cref="Y" /> field.</param>
+    public Vector2D(T x, T y);
+
+    /// <summary>Constructs a vector from the given <see cref="ReadOnlySpan{T}" />. The span must contain at least 4 elements.</summary>
+    /// <param name="values">The span of elements to assign to the vector.</param>
+    public Vector2D(ReadOnlySpan<T> values);
+
+    /// <summary>Gets a vector whose 4 elements are equal to zero.</summary>
+    /// <value>A vector whose two elements are equal to zero (that is, it returns the vector <c>(0,0)</c>.</value>
+    public static Vector2D<T> Zero => new(T.Zero);
+
+    /// <summary>Gets a vector whose 4 elements are equal to one.</summary>
+    /// <value>Returns <see cref="Vector2D{T}" />.</value>
+    /// <remarks>A vector whose two elements are equal to one (that is, it returns the vector <c>(1,1)</c>.</remarks>
+    public static Vector2D<T> One => new(T.One);
+
+    /// <summary>Gets the vector (1,0)).</summary>
+    /// <value>The vector <c>(1,0)</c>.</value>
+    public static Vector2D<T> UnitX => new(T.One, T.Zero);
+
+    /// <summary>Gets the vector (0,1)).</summary>
+    /// <value>The vector <c>(0,1)</c>.</value>
+    public static Vector2D<T> UnitY => new(T.Zero, T.One);
+
+    public ReadOnlySpan<T> Components { get; }
+
+    static Vector2D<T> IVector<Vector2D<T>, T>.CreateFromRepeatingComponent(T scalar);
+
+    public T this[int index] { get; }
+
+    #region Operators
+    /// <summary>Adds two vectors together.</summary>
+    /// <param name="left">The first vector to add.</param>
+    /// <param name="right">The second vector to add.</param>
+    /// <returns>The summed vector.</returns>
+    /// <remarks>The <see cref="op_Addition" /> method defines the addition operation for <see cref="Vector2D{T}" /> objects.</remarks>
+    public static Vector2D<T> operator +(Vector2D<T> left, Vector2D<T> right);
+
+    /// <summary>Subtracts the second vector from the first.</summary>
+    /// <param name="left">The first vector.</param>
+    /// <param name="right">The second vector.</param>
+    /// <returns>The vector that results from subtracting <paramref name="right" /> from <paramref name="left" />.</returns>
+    /// <remarks>The <see cref="op_Subtraction" /> method defines the subtraction operation for <see cref="Vector2D{T}" /> objects.</remarks>
+    public static Vector2D<T> operator -(Vector2D<T> left, Vector2D<T> right);
+
+    /// <summary>Negates the specified vector.</summary>
+    /// <param name="value">The vector to negate.</param>
+    /// <returns>The negated vector.</returns>
+    /// <remarks>The <see cref="op_UnaryNegation" /> method defines the unary negation operation for <see cref="Vector2D{T}" /> objects.</remarks>
+    public static Vector2D<T> operator -(Vector2D<T> value);
+
+    /// <summary>Returns a new vector whose values are the product of each pair of elements in two specified vectors.</summary>
```

Comment:
Nit: `pair` has a specific meaning in vector contexts, you probably want `element-wise` instead.

Reply:
This summary is lifted straight from System.Numerics. That's your cue to open a ticket :P

Reply:
😂. That’s great. Definitely needs fixing